rtt/transports/mqueue/MQSendRecv.cpp

```diff
@@ -75,11 +76,14 @@ void MQSendRecv::setupStream(base::DataSourceBase::shared_ptr ds, base::PortInte
     marshaller_cookie = mtransport.createCookie();
     mis_sender = is_sender;
 
-    std::stringstream namestr;
-    namestr << '/' << port->getInterface()->getOwner()->getName() << '.' << port->getName() << '.' << this << '@' << getpid();
-
     if (policy.name_id.empty())
-        policy.name_id = namestr.str();
+    {
+        std::stringstream name_stream;
+        name_stream << port->getInterface()->getOwner()->getName() << '.' << port->getName() << '.' << this << '@' << getpid();
```

Comment:
Not really related to this PR, but in general a port does not need to have a `DataFlowInterface` and `getInterface()` can return a null pointer. Additionally, but more unlikely, a `DataFlowInterface` does not need to be owned by a `TaskContext` and `getOwner()` can also return null. We already observed segfaults because of this at other places. These are all public and documented RTT classes and we should not assume that ports are only used inside components.

Not sure if for this special case it is guaranteed that none of the pointers is invalid, but maybe it is a good idea to also add a check together and eventually omit the component part in the MQ name?